app/tasks/schedule.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+global $cli;
+global $register;
+
+use Cron\CronExpression;
+use Utopia\App;
+use Utopia\CLI\Console;
+use Utopia\Database\DateTime;
+use Utopia\Database\Query;
+use Swoole\Timer;
+
+const FUNCTION_VALIDATION_TIMER = 180; //seconds
+const FUNCTION_ENQUEUE_TIMER = 60; //seconds
+const ENQUEUE_TIME_FRAME = 60 * 5; // 5 min
+sleep(4); // Todo prevent PDOException
+
+$cli
+->task('schedule-new')
+->desc('Function scheduler task')
+->action(function () use ($register) {
+    Console::title('Scheduler V1');
+    Console::success(APP_NAME . ' Scheduler v1 has started');
+
+    $createQueue = function () use (&$functions, &$queue) {
+        /**
+         * Creating smaller functions list containing 5-min timeframe.
+         */
+        $timeFrame = DateTime::addSeconds(new \DateTime(), ENQUEUE_TIME_FRAME);
+        foreach ($functions as $function) {
+            $cron = new CronExpression($function['schedule']);
+            $next = DateTime::format($cron->getNextRunDate());
+            if ($next < $timeFrame) {
+                $queue[$next][$function['scheduleId']] = $function;
+            }
+        }
+    };
+
+    $removeFromQueue = function ($scheduleId) use (&$queue) {
+        foreach ($queue as $slot => $schedule) {
+            foreach ($schedule as $function) {
```

Comment:
Why do we even use the 2nd loop if we have a nested structure with the function ID as the key? Couldn't we just use `array_key_exists`?